File: jaros/llm/adapters/uppercase_adapter.py
"""Deterministic uppercasing adapter implementing :class:`LlmClient`.

A second concrete adapter (EXT-004 / REQ-2, REQ-4) used to prove that swapping
the configured provider changes output at the same call site with no caller
change. Deterministic and data-only (EXT-004 / REQ-3).
"""

from __future__ import annotations

from dataclasses import dataclass

from jaros.llm.client import LlmRequest, LlmResponse


# #EXT-004-REQ-2 Start
@dataclass(frozen=True)
class UppercaseAdapter:
    """Deterministic adapter that uppercases the prompt; conforms to LlmClient.

    Distinct from :class:`DefaultAdapter` so a config-only swap is observable.
    Holds no provider handles and performs no side effects.
    """

    model: str = "uppercase-echo"

    def complete(self, req: LlmRequest) -> LlmResponse:
        """Return the uppercased prompt as data-only output."""
        return LlmResponse(text=req.prompt.upper(), model=self.model)
# #EXT-004-REQ-2 End

File: src/llm/adapters/uppercase-adapter.ts
import type { LlmClient, LlmRequest, LlmResponse } from "../client";

// #EXT-004-REQ-2 Start
/**
 * A second deterministic, dependency-free `LlmClient`. It exists to prove the
 * boundary is genuinely pluggable: swapping `config.provider` from "default" to
 * "uppercase" changes observable behavior through the SAME `LlmClient.complete`
 * call site, with zero caller changes ([EXT-004 / REQ-2, REQ-4]).
 *
 * Like every adapter it returns DATA ONLY and never drives control flow
 * ([PRIME-001 / REQ-4], EXT-004 REQ-3).
 */
export class UppercaseAdapter implements LlmClient {
  /** Stable identifier reported on every response's `model` field. */
  static readonly model = "uppercase-echo";

  async complete(req: LlmRequest): Promise<LlmResponse> {
    return {
      text: req.prompt.toUpperCase(),
      model: UppercaseAdapter.model,
    };
  }
}
// #EXT-004-REQ-2 End
